jsettlers.logic/src/main/java/jsettlers/input/GuiTaskExecutor.java

```diff
@@ -227,6 +228,10 @@ public void executeTask(TaskPacket iTask) {
 			case UNLOAD_FERRY:
 				unloadFerry((MovableGuiTask) guiTask);
 				break;
+				
+			case SET_MOVEABLE_RATIO:
+				SetMoveableRatioTask task = (SetMoveableRatioTask) guiTask;
```

Comment:
You could either do something like grid.getMovableRatio()+task.getChange() or grid.changeMovableRatio().
I am not sure what would be the best solution.

Reply:
I think in this case nothing is wrong with using a absolute value. A set call is much more easy to understand and to debug then a add call.

Reply:
But an add call is the natural way of translation the arrow symbol to logic calls.
And there seams to be not debugging issues with SPEED_SLOWER and SPEED_FASTER and they don't need timer stuff in the ui to function as intended.

Reply:
The timer stuff should have the same effect with a absolute and a relative value.
With debugging issue i mean not the debugging with a tool. Instead i mean debugging with the brain (understanding and tracking the code).
If i have something like setX(10) then i know the value will be 10. If i have something like increament(10) then i dont know if the result is 10 or 20 or -424152. It could be everything depending on the previous state. Also it would be necessary to have decrement method, too. So the code will be much more robust with the setter.

Reply:
I thought more in terms of ChangeMovableRatioTask and INCREASE_MOVABLE_RATIO and DECREASE_MOVABLE_RATIO task types/actions.

The ratios should obviously be limited between 0 and 100.

If you really want use an absolute value you should implement it via an barfill and not as arrows.